extended/src/main/java/io/kubernetes/client/extended/generic/GenericKubernetesApi.java

```diff
@@ -0,0 +1,680 @@
+ * the custom resources including {get, list, watch, create, update, patch, delete}.
+ *
+ * @param <ApiType> the api type parameter
+ * @param <ApiListType> the api list type parameter
+ */
+public class GenericKubernetesApi<ApiType, ApiListType> {
+
+  // TODO(yue9944882): supports status operations..
+  // TODO(yue9944882): supports generic sub-resoruce operations..
+  // TODO(yue9944882): supports delete-collections..
+
+  private Class<ApiType> apiTypeClass;
+  private Class<ApiListType> apiListTypeClass;
+  private String apiGroup;
+  private String apiVersion;
+  private String resourcePlural;
+  private CustomObjectsApi customObjectsApi;
+
+  /**
+   * Instantiates a new Generic kubernetes api.
+   *
+   * @param apiTypeClass the api type class
+   * @param apiListTypeClass the api list type class
+   * @param apiGroup the api group
+   * @param apiVersion the api version
+   * @param resourcePlural the resource plural
+   */
+  public GenericKubernetesApi(
+      Class<ApiType> apiTypeClass,
+      Class<ApiListType> apiListTypeClass,
+      String apiGroup,
+      String apiVersion,
+      String resourcePlural) {
+    this(
+        apiTypeClass,
+        apiListTypeClass,
+        apiGroup,
+        apiVersion,
+        resourcePlural,
+        new CustomObjectsApi(Configuration.getDefaultApiClient()));
+  }
+
+  /**
+   * Instantiates a new Generic kubernetes api.
+   *
+   * @param apiTypeClass the api type class, e.g. V1Job.class
+   * @param apiListTypeClass the api list type class e.g V1JobList.class
+   * @param apiGroup the api group
+   * @param apiVersion the api version
+   * @param resourcePlural the resource plural, e.g. "jobs"
+   * @param apiClient the api client
+   */
+  public GenericKubernetesApi(
+      Class<ApiType> apiTypeClass,
+      Class<ApiListType> apiListTypeClass,
+      String apiGroup,
+      String apiVersion,
+      String resourcePlural,
+      ApiClient apiClient) {
+    this(
+        apiTypeClass,
+        apiListTypeClass,
+        apiGroup,
+        apiVersion,
+        resourcePlural,
+        new CustomObjectsApi(apiClient));
+  }
+
+  /**
+   * Instantiates a new Generic kubernetes api with the ApiClient specified.
+   *
+   * @param apiTypeClass the api type class, e.g. V1Job.class
+   * @param apiListTypeClass the api list type class e.g V1JobList.class
+   * @param apiGroup the api group
+   * @param apiVersion the api version
+   * @param resourcePlural the resource plural, e.g. "jobs"
+   * @param customObjectsApi the custom objects api
+   */
+  public GenericKubernetesApi(
+      Class<ApiType> apiTypeClass,
+      Class<ApiListType> apiListTypeClass,
+      String apiGroup,
+      String apiVersion,
+      String resourcePlural,
+      CustomObjectsApi customObjectsApi) {
+    if ("".equals(apiGroup)) {
+      throw new IllegalArgumentException(
+          "core group (\"\") not supported, use CoreV1Api otherwise");
```

Comment:
should this say `use "v1" instead`

Reply:
updated to `core group (\"v1\") api not supported, use CoreV1Api instead`, does that sound better?